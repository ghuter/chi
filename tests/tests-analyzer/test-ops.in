f :: fun () {
	x := 0;
	y :: false;

	a :: x == 0;
	b :: x < 10;

	x += 10;
}
