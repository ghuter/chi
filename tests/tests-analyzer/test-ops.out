SFUN(f, {SSEQ(SDECL(x, ECSTI(0)) : i32, SSEQ(const SDECL(y, EFALSE) : bool, SSEQ(const SDECL(a, EBINOP(EQUAL, EMEM(x) : i32, ECSTI(0)) : bool) : bool, SSEQ(const SDECL(b, EBINOP(LT, EMEM(x) : i32, ECSTI(10)) : bool) : bool, SASSIGN(x, EBINOP(ADD, EMEM(x) : i32, ECSTI(10)) : i32)))))}) : void
